sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/NestedColumnAliasing.scala

```diff
@@ -372,6 +372,13 @@ object GeneratorNestedColumnAliasing {
                 e.withNewChildren(Seq(extractor))
             }
 
+            val invalidExtractor = rewrittenG.generator.children.head.collect {
```

Comment:
Could you add some comments for this logic?

Reply:
Sure. I will add some.